Added  Release Checklist for Homepage (#130)
nf-core/nf-co.re: markdown/developers/adding_pipelines.md

* Create a Pull Request including only these two changes against the `dev` branch of the pipeline you're working on

This way, a review process will be very fast and we can merge the changes into the `dev` branch, updating the Docker Image for that pipeline automatically. After ~30 Minutes, the Docker Image for that pipeline is then updated, and you can open your Pull Request containing your actual pipeline code changes.

### Release Checklist

> Whenever a pipeline is considered ready for a (first) release, developers can simply follow these steps to make sure that the pipeline meets the requirements of the nf-core community. The respective pipeline should follow the Guidelines described above in the Guidelines in general. Then developers can follow these steps to get the pipeline to release:

Reviewer: ```suggestion
As described above, when your pipeline is ready for a (first) release please follow the steps below to ensure that the pipeline meets the requirements of the nf-core community:
```

> * Bump the `dev` branch containing the changes for the release to a release version (e.g. 1.0.0): `nf-core bump-version 1.0.0`

Reviewer: ```suggestion
* Bump the version number of the `dev` branch to a release version (e.g. 1.0.0dev > 1.0.0): `nf-core bump-version 1.0.0`
```

> * Check for pipeline dependencies that are out of date and update these accordingly in the `dev` branch: `nf-core lint .` will tell you which ones are outdated via automated API calls to (bio-) conda

Reviewer: ```suggestion
* If applicable, update the software dependencies for the pipeline on the `dev` branch: `nf-core lint .` will tell you which ones are outdated via automated API calls to (bio-) conda
```

* Update the `CHANGELOG`, listing everything that has been added/fixed in this release

> * [Open a Pull Request (PR)](https://help.github.com/en/articles/creating-a-pull-request) from `dev` to `master` on GitHub after adjusting all of this - waiting for tests to pass

Reviewer: ```suggestion
* [Open a Pull Request (PR)](https://help.github.com/en/articles/creating-a-pull-request) from `dev` to `master` on GitHub after adjusting all of this and wait for all of the tests to pass
```

> * Link some reviewers (2 are required for merging to `master`) - asking for a final review on your release

Reviewer: ```suggestion
* Request some reviewers (2 are required for merging to `master`) asking for a final review before your release
```

* Once approved by two reviewers, merge your PR into `master`
* Go to GitHub and [create a new release for your pipeline](https://help.github.com/en/articles/creating-releases)
  * Please make sure to use strictly numeric release numbers, most people follow [Semantic Versioning](https://semver.org/), e.g. 1.0.0, 1.0.
* Optional: Use a [nice code name](http://www.codenamegenerator.com/) for your pipeline release

> * Create your release - tests will automatically run, DockerHub will generate a tagged container for that release.

Reviewer: ```suggestion
* Create your release - tests will run automatically and DockerHub will generate a tagged container for that release.
```

Reviewer: Looks good @apeltzer . Maybe need something about Zenodo IDs in the sections above and/or here i.e. need to make sure that the pipeline is linked on GitHub via WebHooks and then will need to create an additional PR to `master` via a branch to add the Zenodo ID by commenting out the code in the `Citation` section in the template.

Author: Very true! I'll add this now, too!
